clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -97,12 +100,44 @@ public static String randomString(int len) {
     }
 
     /**
-     * Creates an empty file in the default temporary-file directory, using `kafka` as the prefix and `tmp` as the
+     * Create an empty file in the default temporary-file directory, using `kafka` as the prefix and `tmp` as the
      * suffix to generate its name.
      */
     public static File tempFile() throws IOException {
-        File file = File.createTempFile("kafka", ".tmp");
+        final File file = File.createTempFile("kafka", ".tmp");
```

Comment:
Where is the inner class? I don't see it in my diff. You removed the shutdown hook in this version right?